server/src/test/java/io/unitycatalog/server/utils/OptionParserTest.java

```diff
@@ -0,0 +1,72 @@
+package io.unitycatalog.server.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import lombok.Getter;
+import lombok.Setter;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OptionParserTest {
+  private final PrintStream originalOut = System.out;
+
+  private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+  @Setter
+  @Getter
+  class NoExitOptionsParser extends OptionParser {
+    int exitCode = -128;
+
+    @Override
+    protected void exit(int code) {
+      setExitCode(code);
+    }
+  }
+
+  @BeforeEach
+  void setUp() {
+    System.setOut(new PrintStream(out));
+  }
+
+  @AfterEach
+  void tearDown() {
+    System.setOut(originalOut);
+  }
+
+  @Test
+  void testParseCLIOptions() {
+    OptionParser optionParser = new OptionParser();
+    optionParser.parse(new String[] {"-p", "8081"});
+    assertEquals(8081, optionParser.getPort());
+  }
+
+  @Test
+  void testParseCLIOptionsWithVersion() {
+    NoExitOptionsParser optionParser = new NoExitOptionsParser();
+    optionParser.parse(new String[] {"-v"});
+    assertEquals(0, optionParser.getExitCode());
+    assertEquals(VersionUtils.VERSION, out.toString().trim());
+  }
+
+  @Test
+  void testParseCLIOptionsWithHelp() {
+    NoExitOptionsParser optionParser = new NoExitOptionsParser();
+    optionParser.parse(new String[] {"-h"});
+    assertEquals(0, optionParser.getExitCode());
+    assertTrue(out.toString().contains("bin/start-uc-server"));
+    assertTrue(out.toString().contains("Print help message."));
+  }
+
+  @Test
+  void testParseCLIOptionsWithInvalidOption() {
+    NoExitOptionsParser optionParser = new NoExitOptionsParser();
+    optionParser.parse(new String[] {"-x"});
+    assertEquals(-1, optionParser.getExitCode());
+    assertTrue(out.toString().contains("Parsing Failed. Reason: Unrecognized option: -x"));
+    assertTrue(out.toString().contains("Print help message."));
```

Comment:
can we change the asserts here to use assertJ, we're trying to use assertJ across the project going forward.